modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -400,7 +399,17 @@ public void parse(DocumentParserContext context) throws IOException {
         configureContext(executionContext, isMapUnmappedFieldAsText());
 
         XContentParser parser = context.parser();
-        QueryBuilder queryBuilder = parseQueryBuilder(parser, parser.getTokenLocation());
+        QueryBuilder queryBuilder;
+        try {
+            //make sure that we don't expand dots in field names while parsing, otherwise queries will
+            // fail parsing due to unsupported inner objects
+            context.path().setWithinLeafObject(true);
+            queryBuilder = parseTopLevelQuery(parser);
+        } catch (IOException e) {
+            throw new ParsingException(parser.getTokenLocation(), "Failed to parse", e);
+        } finally {
+            context.path().setWithinLeafObject(false);
+        }
```

Comment:
This is an interesting one: I got test failures that percolator queries that target field with dots in their names can no longer be parsed, because we expand dots while parsing. I was surprised that this was not an issue before, but the reason is that when parsing named objects, we were parsing using the inner parser and not the wrapper, hence queries were not getting their dots expanded.

As I ensured that we preserve the parser when parsing the named object, I had to address this issue in the percolator field mapper by disabling dots expansion explicitly for queries, like we do under subobjects:false.